### PR TITLE
Support query option schema

### LINKED ISSUE #437: [BUG] Loading data from vertical using the 'query' option always uses the default schema
## Environment
- Spark version: 3.2
- Hadoop version: 3.3.1
- Vertica version: 10.1
- Vertica Spark Connector version: 3.2.1
- Java version:
- Additional Environment Information:

---

## Problem Description
When loading vertica data into a dataframe using the 'query' option, the 'dbschema' seems to be ignored. Prefixing the SQL tables with the schema works as expected, but this is not always convenient. In my case, I am getting the SQL from a semantic layer, which doesn't add the scheme to the table names and expects the correct schema to be configured as default.

1. Steps to reproduce:

Use spark.read.format, and define the 'query' and 'dbschema' options. Set 'dbschema' to a schema other than the default. Use a SQL query that doesn't prepend schema to the tablenames and that targets a table that only exists on dbschema. The call incorrectly fails.

2. Expected behaviour:

The 'dbschema' option should be honored when 'query' is used

3. Actual behaviour:

'dbschema' is not honored

5. Error message/stack trace:
6. Code sample or example on how to reproduce the issue:

---

## Spark Connector Logs
- Add related logs entries here.

### PR DESCRIPTION
**Pull request by Aryex**
### Summary

Adding support for using `query` option with `dbschema` option.

### Description

With both defined, the schema specified in `dbschema` will be appended to the table name in the query's FROM clause. This will also work when string literals are used.
The `dbschema` _will not be appended_ if:

1. The FROM clause is a query.
2. The FROM clause already has a schema defined.

For reviewing, main code logic starts at `SchemaTools.addDbSchemaToQuery()`.

### Related Issue

Close #437 

### Additional Reviewers

@jeremyp-bq 
@jonathanl-bq 
@alexey-temnikov 
@alexr-bq

### CODE REVIEW

File: connector/src/main/scala/com/vertica/spark/util/schema/SchemaTools.scala [hunk 1]
```diff
@@ -682,6 +692,89 @@ class SchemaTools(ctTools: ComplexTypesSchemaTools = new ComplexTypesSchemaTools
     // Use recursion to break early
     findEmptyColumnName(schema.fields.toList)
   }
+
+  private def noDotOutsideStringLiteral(str: String): Boolean = {
```
Comment (jeremyp-bq) [inline connector/src/main/scala/com/vertica/spark/util/schema/SchemaTools.scala:696]: Should add a scaladoc to a few functions here.  Also, can we improve the naming of a few of these functions?

For example:
* noDotOutsideStringLiteral -> containsSchema
* dotOutsideStringLiteral -> containsSchemaHelper
* recursion -> addDbSchemaToQueryHelper

Comment (Aryex) [reply connector/src/main/scala/com/vertica/spark/util/schema/SchemaTools.scala]: Updated. Regex is used instead to replace the noDotOutsideStringLiteral(). Functional names are also updated.

Comment (Aryex) [reply connector/src/main/scala/com/vertica/spark/util/schema/SchemaTools.scala]: about the scaladoc, do you mean for `addDbSchemaToQuery`? It is defined at the trait definition.

Comment (jeremyp-bq) [reply connector/src/main/scala/com/vertica/spark/util/schema/SchemaTools.scala]: That's sufficient.

**Review by jeremyp-bq — APPROVED** [review]

**Pull request merged**